pkg/auth/oauth/external/handler.go

```diff
@@ -145,7 +145,7 @@ func DefaultState() State {
 func (defaultState) Generate(w http.ResponseWriter, req *http.Request) (string, error) {
 	state := url.Values{
 		"csrf": {"..."}, // TODO: get csrf
-		"then": {req.URL.String()},
+		"then": {url.QueryEscape(req.URL.String())},
```

Comment:
I was having an issue where the URL was not encoded and part of "then" was being processed through the main query, specifically request_type was not part of "then" but was seen as a separate parameter (e.g. state, then, request_type)